src/Bicep.Core/Registry/Oci/OciArtifactResult.cs

```diff
@@ -33,17 +33,9 @@ public OciArtifactResult(BinaryData manifestBits, string manifestDigest, IEnumer
 
         public IEnumerable<OciArtifactLayer> Layers { get; init; }
 
-        public BinaryData GetSingleLayerByMediaType(string mediaType)
+        public BinaryData? TryGetSingleLayerByMediaType(string mediaType)
```

Comment:
```suggestion
        // Expects either zero or exactly one layer of the given media type
        public BinaryData? TryGetSingleLayerByMediaType(string mediaType)
```

Reply:
Okay, I see you changed that.  The intention was that we want to throw if there are more layers of a given type than we expect, because it means we don't know how to handle that (likely the versioning should have changed).  (This is different from there being an optional layer of a different type.). We certainly don't know how to handle two main.json files.

Reply:
see other comment.  Thoughts?

Reply:
Understood, reversed the change.